Running dotnet pack with Paket.Restore.Targets referenced in a new csproj file with msbuild 16 fails.

### Description

Running dotnet pack with Paket.Restore.Targets referenced in a new csproj file with msbuild 16 fails.

```
C:\code\xyz\src\.paket\Paket.Restore.targets(298,15): error MSB4064: The "BuildOutputFolder" parameter is not supported by the "PackTask" task. Verify the parameter exists on the task, and it is a settable public instance property. [C:\code\.....csproj]
C:\code\xyz\src\.paket\Paket.Restore.targets(261,5): error MSB4063: The "PackTask" task could not be initialized with its input parameters.  [C:\code\.....csproj]
```

### Known workarounds

Dont install msbuild 16 on the system (luckily I had not gotten around to installing it on the build servers yet) :D 
I have no idea if it is possible to tell dotnet to use a specific version of msbuild, but I havent figured it out so far...but if that is possible that is probably an option as well.

--- a/src/Paket.Core/embedded/Paket.Restore.targets
+++ b/src/Paket.Core/embedded/Paket.Restore.targets
@@ -245,7 +245,7 @@
       <UseMSBuild15_8_Pack>false</UseMSBuild15_8_Pack>
       <UseMSBuild15_8_Pack Condition=" '$(NuGetToolVersion)' != '4.0.0' AND (! $(UseMSBuild15_9_Pack)) AND (! $(UseMSBuild16_0_Pack)) ">true</UseMSBuild15_8_Pack>
       <UseNuGet4_Pack>false</UseNuGet4_Pack>
-      <UseNuGet4_Pack Condition=" (! $(UseMSBuild15_8_Pack)) AND (! $(UseMSBuild15_9_Pack)) ">true</UseNuGet4_Pack>
+      <UseNuGet4_Pack Condition=" (! $(UseMSBuild15_8_Pack)) AND (! $(UseMSBuild15_9_Pack)) AND (! $(UseMSBuild16_0_Pack)) ">true</UseNuGet4_Pack>
       <AdjustedNuspecOutputPath>$(BaseIntermediateOutputPath)$(Configuration)</AdjustedNuspecOutputPath>
       <AdjustedNuspecOutputPath Condition="@(_NuspecFilesNewLocation) == ''">$(BaseIntermediateOutputPath)</AdjustedNuspecOutputPath>
     </PropertyGroup>

